### PR TITLE
Program Editor Screen

### PR DESCRIPTION
**Pull request by JoshuaSchaefer**
Add a new screen to show the program name, description, and credits editor.
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
Currently, the program editor is shown on the program page. However, this PR would move the editor to its own screen. In future PRs we will show text fields for the name, description, and credits sections at the same time on this new screen. This will make it easier for users to add and manage custom translations instead of adding translations for each section separately.

### Tests - additional information
There are not any new tests for this PR, but some minor updates to existing tests were made to reflect the changes.

### CODE REVIEW

File: assets/styles/layout/sidebar.scss [hunk 1]
```diff
@@ -92,6 +92,11 @@ $sidebar-width: 250px;
       left: $sidebar-width;
       width: calc(100vw - #{$sidebar-width});
     }
+
+    .full-screen {
```
Comment (danielzhang130) [inline assets/styles/layout/sidebar.scss:96]: how is this used?

**Review by danielzhang130 — COMMENTED** [review]
Overall, really pleased how simple the changes are

File: assets/styles/layout/sidebar.scss (hunk 1 above)
Comment (JoshuaSchaefer) [reply assets/styles/layout/sidebar.scss:96]: It practically reduces the screen size when the left sidebar is constantly open when it is on a bigger page. Without it, the text editor is behind the left sidebar. 
Screen Shot 2022-02-07 at 5 03 18 PM

**Comment by JoshuaSchaefer** [comment]
> Overall, really pleased how simple the changes are

Yeah, I thought it was going to be a lot more work, but I was practically able to copy over the project list code and styling.

**Review by danielzhang130 — APPROVED** [review]
Should we give them a heads up to what's coming overall? like this is part of a bigger redesign. It's also important that they share the same vision of the final design.

**Comment by JoshuaSchaefer** [comment]
> 

Yeah, I was wondering if I should add the mockup document to the PR or do you think just an explanation of what is coming is okay? Or should we message them on slack about it first?

**Comment by danielzhang130** [comment]
I think we should message on slack without any code first. I think they've seen the design before, but just to remind them

**Pull request closed without merging**